processors/enduserprocessor/enduserprocessor.go

```diff
@@ -0,0 +1,534 @@
+		for j := 0; j < ilss.Len(); j++ {
+			ils := ilss.At(j)
+
+			spans := ils.Spans()
+			for k := 0; k < spans.Len(); k++ {
+				span := spans.At(k)
+
+				span.Attributes().ForEach(func(k string, v pdata.AttributeValue) {
+					unindexedKey := piifilterprocessor.UnindexedKey(k)
+					if configs, ok := p.attributeConfigMap[unindexedKey]; ok {
+						for _, cfg := range configs {
+							p.capture(span, cfg, v.StringVal())
+						}
+					}
+				})
+			}
+		}
+	}
+
+	return traces, nil
+}
+
+func (p *processor) capture(span pdata.Span, enduser config, value string) {
+	if !p.passesConditions(span, enduser.Conditions) {
+		return
+	}
+
+	var u *user
+	switch enduser.Type {
+	case "id":
+		u = p.idCapture(value)
+	case "role":
+		u = p.roleCapture(value)
+	case "scope":
+		u = p.scopeCapture(value)
+	case "authheader":
+		u = p.authHeaderCapture(enduser, value)
+	case "json":
+		u = p.jsonCapture(enduser, value)
+	case "urlencoded":
+		u = p.urlencodedCapture(enduser, value)
+	case "cookie":
+		u = p.cookieCapture(enduser, value)
+	default:
+		p.logger.Warn("Unknown enduser type, skipping", zap.String("type", enduser.Type), zap.String("key", enduser.AttributeKey))
+	}
+
+	if u == nil {
+		return
+	}
+
+	if len(u.id) > 0 {
+		addSpanAttribute(span, enduserIDAttribute, u.id)
+	}
+
+	if len(u.role) > 0 {
+		addSpanAttribute(span, enduserRoleAttribute, u.role)
+	}
+
+	if len(u.scope) > 0 {
+		addSpanAttribute(span, enduserScopeAttribute, u.scope)
+	}
+
+	if len(u.session) > 0 {
+		addSpanAttribute(span, enduserSessionAttribute, u.session)
+	}
+}
+
+func addSpanAttribute(span pdata.Span, key string, value string) {
+	attributes := span.Attributes()
+	// don't overwrite existing attributes
+	if _, ok := attributes.Get(key); ok {
+		return
+	}
+	attributes.Insert(key, pdata.NewAttributeValueString(value))
+}
+
+// only capture the user info if all conditions
+// are true.  If an conditions key does not exist
+// in the span, that is considered a failed condition
+func (p *processor) passesConditions(span pdata.Span, conditions []Condition) bool {
+	attribMap := span.Attributes()
+
+	for _, condition := range conditions {
+		attrib, ok := attribMap.Get(condition.Key)
+		if !ok {
+			return false
+		}
+
+		value := attrib.StringVal()
+		if value == "" {
+			return false
+		}
+
+		matched, err := regexp.MatchString(condition.Regex, value)
+		if err != nil {
+			p.logger.Warn("Could not evaluate enduser condition", zap.Error(err))
+			return false
+		}
+		if !matched {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (p *processor) idCapture(value string) *user {
+	return &user{id: value}
+}
+
+func (p *processor) roleCapture(value string) *user {
+	return &user{role: value}
+}
+
+func (p *processor) scopeCapture(value string) *user {
+	return &user{scope: value}
+}
+
+func (p *processor) authHeaderCapture(enduser config, value string) *user {
+	lcValue := strings.ToLower(value)
+	if strings.HasPrefix(lcValue, "bearer ") {
```

Comment:
how about encapsulating the logic inside this ifs with meaningful function/methods?